blues/sparrow-reference-web-app
# feat: Add Sensor Config API endpoint

# Problem Context
We currently have stubbed API endpoints for all of the NextJS API routes which return dummy JSON. We want to refactor those endpoints to hit the Notehub API and to return real data.

## Changes
This PR adds a live Sensor Config API endpoint which returns the `config.db` file for a given Sensor ID. Additionally, the `config.db` file can be set at the same endpoint via POST with a JSON body.

## Testing

You will need to add an additional ENV var to run this branch: `HUB_PRODUCT_UID` which for the sparrow project will be `net.ozzie.ray:sparrow`

Additionally, you'll want to add `HUB_SENSOR_MAC=20323746323650050028000a` (or another sensor ID) to your `.env.test.local` so the handler has a macAddress to test with.

To test:
- Run `yarn test` to run Unit tests. You should see nothing but green :heavy_check_mark: 
- Alternatively you can visit the new API endpoint in your browser: `/api/gateway/dev:868050040065365/sensor/20323746323650050028000a/config`

## Ticket(s)
https://www.pivotaltracker.com/story/show/180562817
